scripts/consumer_apps_builder.py

```diff
@@ -0,0 +1,103 @@
+
+import os
+import subprocess 
+import xml.etree.ElementTree as ET
+
+# Script to build the .apks for all consumer apps, off of the latest release build of CommCare on jenkins
+# All paths are written assuming this script will be run from the PARENT directory of commcare-odk/,
+# unless otherwise specified 
+
+
+# Path to the directory where all user-provided information and files for each consumer app lives. 
+# This directory should store a list of directories, 1 for each consumer app we are building.
+# The expected format for a single app directory is as follows:
+# - config.txt: a text file containing the following 5 pieces of information in order, on a single comma-separated line, e.g: app_id,app_domain,build_number,username,password
+# - ic_launcher.zip: a zip file generated from Android Asset Studio of the desired app icon (MUST have this exact name)
+PATH_TO_STATIC_RESOURCES_DIR = "./consumer-apps-resources"
+
+# Path to the commcare-odk app directory
+PATH_TO_ODK_DIR = "./commcare-odk/"
+
+# Path to the standalone directory
+PATH_TO_STANDALONE_DIR = PATH_TO_ODK_DIR + "app/standalone/"
+
+# Path to the directory where all app assets should be placed, RELATIVE to the commcare-odk/
+# directory, since we have cd'ed into that directory at the time this is used
+PATH_TO_ASSETS_DIR_FROM_ODK = "./app/standalone/assets/"
+
+
+def checkout_or_update_static_resources_repo():
+	if not os.path.exists(PATH_TO_STATIC_RESOURCES_DIR):
+		subprocess.call(["git", "clone", "https://github.com/dimagi/consumer-apps-resources"])
+	os.chdir(PATH_TO_STATIC_RESOURCES_DIR)
+	subprocess.call(["git", "pull"])
+	os.chdir('../')
+
+
+def build_apks_from_resources():
+	for (app_dir_name, sub_dir_list, files_list) in os.walk(PATH_TO_STATIC_RESOURCES_DIR):
+		if '.git' not in app_dir_name and app_dir_name != PATH_TO_STATIC_RESOURCES_DIR:
+			build_apk_from_directory_contents(app_dir_name, files_list)
+
+
+def build_apk_from_directory_contents(app_sub_dir, files_list):
+	full_path_to_config_file = os.path.join(app_sub_dir, files_list[0])
```

Comment:
For future PR: might want to somehow check that the first entry is indeed the config file